src/connectionpools.jl

```diff
@@ -242,9 +245,16 @@ created and passed the `max`, `idle_timeout`, and `reuse` keyword arguments if p
 The provided function `f` must create a new connection instance of type `C`.
 The acquired connection MUST be returned to the pool by calling `release(pool, key, conn)` exactly once.
 """
-function acquire(f, pool::Pool{C}, key; forcenew::Bool=false, kw...) where {C}
+function acquire(f, pool::Pool{Connection{Sockets.TCPSocket}}, key; forcenew::Bool=false, kw...)
     pod = lock(pool.lock) do
-        get!(() -> Pod(C; kw...), pool.pods, key)
+        get!(() -> Pod(Connection{Sockets.TCPSocket}; kw...), pool.pods, key)
```

Comment:
i think nothing in ConnectionPools is supposed to be `Connection` specific -- `Connection` is just one implementation on the required interface, but a user could provide an alternative one -- so i think this should stay as is... does it need to change? isn't `C` enough info anyway?